### PR TITLE
Fix locations of attributes docstrings

### PR DESCRIPTION
**Pull request by CunliangGeng**
To solve the issue https://github.com/NPLinker/nplinker/issues/232#issuecomment-2023261162.

The docstrings of attributes are moved to from `__init__` level to class level.

Other changes:
- fix format and static typing errors
- update settings for format and static typing check

### CODE REVIEW

**Comment by CunliangGeng** [comment]
* **#241** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/241?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#240** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/240?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#239** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/239?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CunliangGeng and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pyproject.toml [hunk 1]
```diff
@@ -56,9 +56,12 @@ dev = [
     "coverage[toml]",
     # static typing
     "mypy",
-    "types-toml",
-    "types-Deprecated",
     "typing_extensions",
+    # stub packages 
+    "types-Deprecated",
```
Comment (gcroci2) [inline pyproject.toml:61]: What are these packages for?

**Review by gcroci2 — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (CunliangGeng) [reply pyproject.toml:61]: They are [stub files](https://mypy.readthedocs.io/en/stable/stubs.html) to provide static typing info
> A stub file is a file containing a skeleton of the public interface of that Python module, including classes, variables, functions – and most importantly, their types.

It's required for mypy to check the typings of some dependencies.

**Comment by CunliangGeng** [comment]
### Merge activity

* **Apr 18, 9:13 AM EDT**: @CunliangGeng started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/239).
* **Apr 18, 9:13 AM EDT**: @CunliangGeng merged this pull request with [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/239).

**Pull request merged**